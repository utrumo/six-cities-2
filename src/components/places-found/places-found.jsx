import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import Selectors from '../../store/selectors.js';

const PlacesFound = ({count, city}) => (
  <b className="places__found">{count} places to stay in {city}</b>
);

PlacesFound.propTypes = {
  count: PropTypes.number.isRequired,
  city: PropTypes.string.isRequired
};

const mapStateToProps = (state) => ({
  count: Selectors.getOffersCount(state),
  city: Selectors.getCurrentLocation(state)
});

export {PlacesFound};
export default connect(mapStateToProps)(PlacesFound);
