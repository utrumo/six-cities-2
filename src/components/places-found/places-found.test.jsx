import React from 'react';
import renderer from 'react-test-renderer';
import {PlacesFound} from './places-found.jsx';

it(`PlacesFound renders correctly`, () => {
  const tree = renderer.create(<PlacesFound count={5} city="Moscow" />).toJSON();
  expect(tree).toMatchSnapshot();
});
