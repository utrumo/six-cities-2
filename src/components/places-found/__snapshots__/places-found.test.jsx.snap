// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesFound renders correctly 1`] = `
<b
  className="places__found"
>
  5
   places to stay in 
  Moscow
</b>
`;
